bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/validation/internal/TypeIntrospections.java

```diff
@@ -161,19 +161,26 @@ boolean isAssignable(Object value) {
         }
 
         @Override
-        boolean isMinViolated(Object value, int min) {
+        boolean isMinViolated(Object value, BigDecimal min) {
+            if (min == null) {
+                return false;
+            }
             if (isBigDecimalInstance(value)) {
-                return Float.compare(((BigDecimal) value).floatValue(), min) < 0;
+                return ((BigDecimal) value).compareTo(min) < 0;
             }
-            return Float.compare((float) value, min) < 0;
+            return Float.compare((float) value, Float.valueOf(min.toString())) < 0;
```

Comment:
@ivivanov-bg WDYT about

```Java
final BigDecimal bd;
if (isBigDecimalInstance(value)) {
    bd = (BigDecimal) value;
} else {
    bd = new BigDecimal(value.toString());
}
return bd.compareTo(min) < 0;
```

at least it does not make a big difference. Personally I assume it could be simpler to read.

WDYT?